File: mobile/src/main/java/com/example/steven/stlbusarrivals/model2/PathList.java
package com.example.steven.stlbusarrivals.model2;

import com.google.android.gms.wearable.DataMap;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * Created by Steven on 2016-02-03.
 */
public class PathList extends ArrayList<Path> implements Serializable{

    public DataMap putData(){
        DataMap map = new DataMap();
        for(int i = 0; i<this.size(); i++){
            map.putDataMap("path"+i,this.get(i).putData());
        }
        return map;
    }
}

File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/model/PathList.kt
package com.tilly.steven.stlbusarrivals.model

import com.google.android.gms.wearable.DataMap
import java.io.Serializable
import java.util.*

/**
 * Created by Steven on 2016-02-03.
 */
class PathList : ArrayList<Path>(), Serializable {

    fun putData(): DataMap {
        val map = DataMap()
        for (i in 0 until this.size) {
            map.putDataMap("path$i", this[i].putData())
        }
        return map
    }
}
